### PR TITLE
【映射文档】添加 api 别名表格

### PR DESCRIPTION
**Pull request by RedContritio**
在映射主目录中添加 api 别名表格

**该表格为静态数据，数据依赖 PaConvert，更新时需要手动更新**。

- 表格**更新方法**：
  1. 将 PaConvert 的 `api_alias_mapping.json` 文件复制到 `validate_mapping_in_api_difference.py` 所在目录中
  2. 执行 `validate_mapping_in_api_difference.py`，基于 json 生成 `pytorch_api_mapping_cn.tmp.md` 文件
  3. 在 `pytorch_api_mapping_cn.tmp.md` 找到对应的 `## alias` 部分，复制表格到 `pytorch_api_mapping_cn.md` 中，完成更新。
- **缺失 API**：别名和本名都不存在文档的 API

  > Alias Warning: api `torch.Tensor.atan2_` have no mapping doc, failed to reference from alias `torch.Tensor.arctan2_`
  > Alias Warning: api `torch.amp.autocast_mode.autocast` have no mapping doc, failed to reference from alias `torch.amp.autocast`
  > Alias Warning: api `torch.celu_` have no mapping doc, failed to reference from alias `torch.nn.functional.celu_`
  > Alias Warning: api `torch.channel_shuffle` have no mapping doc, failed to reference from alias `torch.nn.functional.channel_shuffle`
  > Alias Warning: api `torch.conv_tbc` have no mapping doc, failed to reference from alias `torch.nn.functional.conv_tbc`
  > Alias Warning: api `torch.distributions.binomial.Binomial` have no mapping doc, failed to reference from alias `torch.distributions.Binomial`
  > Alias Warning: api `torch.distributions.continuous_bernoulli.ContinuousBernoulli` have no mapping doc, failed to reference from alias `torch.distributions.ContinuousBernoulli`
  > Alias Warning: api `torch.distributions.exponential.Exponential` have no mapping doc, failed to reference from alias `torch.distributions.Exponential`
  > Alias Warning: api `torch.distributions.multivariate_normal.MultivariateNormal` have no mapping doc, failed to reference from alias `torch.distributions.MultivariateNormal`
  > Alias Warning: api `torch.distributions.one_hot_categorical.OneHotCategorical` have no mapping doc, failed to reference from alias `torch.distributions.OneHotCategorical`
  > Alias Warning: api `torch.distributions.transforms.CatTransform` have no mapping doc, failed to reference from alias `torch.distributions.CatTransform`
  > Alias Warning: api `torch.distributions.transforms.CumulativeDistributionTransform` have no mapping doc, failed to reference from alias `torch.distributions.CumulativeDistributionTransform`
  > Alias Warning: api `torch.distributions.transforms.SoftplusTransform` have no mapping doc, failed to reference from alias `torch.distributions.SoftplusTransform`
  > Alias Warning: api `torch.igamma` have no mapping doc, failed to reference from alias `torch.special.gammainc`
  > Alias Warning: api `torch.igammac` have no mapping doc, failed to reference from alias `torch.special.gammaincc`
  > Alias Warning: api `torch.nn.modules.Linear` have no mapping doc, failed to reference from alias `torch.nn.linear`
  > Alias Warning: api `torch.nn.modules.RNNCellBase` have no mapping doc, failed to reference from alias `torch.nn.RNNCellBase`
  > Alias Warning: api `torch.nn.modules.linear.Linear` have no mapping doc, failed to reference from alias `torch.nn.linear`
  > Alias Warning: api `torch.nn.modules.rnn.RNNCellBase` have no mapping doc, failed to reference from alias `torch.nn.RNNCellBase`
  > Alias Warning: api `torch.nn.parameter.UninitializedParameter` have no mapping doc, failed to reference from alias `torch.nn.UninitializedParameter`
  > Alias Warning: api `torch.selu_` have no mapping doc, failed to reference from alias `torch.nn.functional.selu_`
  > Alias Warning: api `torch.threshold_` have no mapping doc, failed to reference from alias `torch.nn.functional.threshold_`
  > Alias Warning: api `torch.torch.int32` have no mapping doc, failed to reference from alias `torch.int32`
  > Alias Warning: api `transformers.generation.logits_process.LogitsProcessorList` have no mapping doc, failed to reference from alias `transformers.generation.LogitsProcessorList`
  > Alias Warning: api `transformers.modeling_utils.PreTrainedModel` have no mapping doc, failed to reference from alias `transformers.PreTrainedModel`
  > Alias Warning: api `transformers.utils.logging.get_logger` have no mapping doc, failed to reference from alias `transformers.logging.get_logger`
  >

### CODE REVIEW

File: docs/guides/model_convert/convert_from_pytorch/pytorch_api_mapping_cn.md [hunk 1]
```diff
@@ -1653,3 +1653,181 @@
 | NOT-IMPLEMENTED-ITEM(`torch.utils.cpp_extension.load_inline`, https://pytorch.org/docs/stable/cpp_extension.html#torch.utils.cpp_extension.load_inline) |
 
 ***持续更新...***
+
+## API 别名映射列表
+
+| Pytorch 最新 release | Paddle develop | 映射关系分类 | 备注 |
+| -------------------- | -------------- | ----------- | ---- |
+| [`torch.Tensor.abs`](https://pytorch.org/docs/stable/generated/torch.Tensor.absolute.html#torch-tensor-absolute) | [`paddle.Tensor.abs`](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/Tensor_cn.html#abs-name-none) | 无参数 | `torch.Tensor.absolute`别名，[详细对比](https://github.com/PaddlePaddle/docs/tree/develop/docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.absolute.md) |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/pytorch_api_mapping_cn.md]: 如果已经有的话就不需要加别名了，不要弄冗余了

Comment (RedContritio) [reply docs/guides/model_convert/convert_from_pytorch/pytorch_api_mapping_cn.md]: 已删除冗余项

File: docs/guides/model_convert/convert_from_pytorch/validate_mapping_in_api_difference.py [hunk 2]
```diff
@@ -571,6 +574,54 @@ def auto_fill_index_from_api_diff(basedir, meta_dict) -> None:
         filled_count += 1
         target[api_type][torch_api] = column
 
+    alias_filename = "api_alias_mapping.json"
+    if os.path.exists(alias_filename) and os.path.isfile(alias_filename):
+        target["alias"] = {}
+        with open(alias_filename, "r", encoding="utf-8") as f:
+            api_alias = json.load(f)
+            for api_name, alias_name in api_alias.items():
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/validate_mapping_in_api_difference.py]: `api_alias_mapping.json`的key是别名，value是推荐名

Comment (RedContritio) [reply docs/guides/model_convert/convert_from_pytorch/validate_mapping_in_api_difference.py]: 因为正反都看到有例子，所以将两边都试着去查找文档，如果一边能找到一边找不到就加入表格。

Comment (zhwesky2010) [reply docs/guides/model_convert/convert_from_pytorch/validate_mapping_in_api_difference.py]: > 因为正反都看到有例子，所以将两边都试着去查找文档，如果一边能找到一边找不到就加入表格。

设计这个mapping的时候是这样的：key为api_alias_name，value是api_name，是根据官网描述来判定的。
但实际写映射文档的同学，可能是 `别名`、`主名` 随机写了映射文档，所以就是两边互找就行，但是变量名称建议按设计的原则来，这样日后可维护性好些

File: docs/guides/model_convert/convert_from_pytorch/pytorch_api_mapping_cn.md [hunk 3]
```diff
@@ -1653,3 +1653,119 @@
 | NOT-IMPLEMENTED-ITEM(`torch.utils.cpp_extension.load_inline`, https://pytorch.org/docs/stable/cpp_extension.html#torch.utils.cpp_extension.load_inline) |
 
 ***持续更新...***
+
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/pytorch_api_mapping_cn.md:1656]: 表头内容：

infoflow 2024-06-05 12-16-48


这个地方也需要改一下，下个PR改一下吧

**Comment by zhwesky2010** [comment]
> 在映射主目录中添加 api 别名表格
> 
> **该表格为静态数据，数据依赖 PaConvert，更新时需要手动更新**。
> 
> * 表格**更新方法**：
>   
>   1. 将 PaConvert 的 `api_alias_mapping.json` 文件复制到 `validate_mapping_in_api_difference.py` 所在目录中
>   2. 执行 `validate_mapping_in_api_difference.py`，基于 json 生成 `pytorch_api_mapping_cn.tmp.md` 文件
>   3. 在 `pytorch_api_mapping_cn.tmp.md` 找到对应的 `## alias` 部分，复制表格到 `pytorch_api_mapping_cn.md` 中，完成更新。
> * **缺失 API**：别名和本名都不存在文档的 API
>   > Alias Warning: api `torch.Tensor.atan2_` have no mapping doc, failed to reference from alias `torch.Tensor.arctan2_`
>   > Alias Warning: api `torch.amp.autocast_mode.autocast` have no mapping doc, failed to reference from alias `torch.amp.autocast`
>   > Alias Warning: api `torch.celu_` have no mapping doc, failed to reference from alias `torch.nn.functional.celu_`
>   > Alias Warning: api `torch.channel_shuffle` have no mapping doc, failed to reference from alias `torch.nn.functional.channel_shuffle`
>   > Alias Warning: api `torch.conv_tbc` have no mapping doc, failed to reference from alias `torch.nn.functional.conv_tbc`
>   > Alias Warning: api `torch.distributions.binomial.Binomial` have no mapping doc, failed to reference from alias `torch.distributions.Binomial`
>   > Alias Warning: api `torch.distributions.continuous_bernoulli.ContinuousBernoulli` have no mapping doc, failed to reference from alias `torch.distributions.ContinuousBernoulli`
>   > Alias Warning: api `torch.distributions.exponential.Exponential` have no mapping doc, failed to reference from alias `torch.distributions.Exponential`
>   > Alias Warning: api `torch.distributions.multivariate_normal.MultivariateNormal` have no mapping doc, failed to reference from alias `torch.distributions.MultivariateNormal`
>   > Alias Warning: api `torch.distributions.one_hot_categorical.OneHotCategorical` have no mapping doc, failed to reference from alias `torch.distributions.OneHotCategorical`
>   > Alias Warning: api `torch.distributions.transforms.CatTransform` have no mapping doc, failed to reference from alias `torch.distributions.CatTransform`
>   > Alias Warning: api `torch.distributions.transforms.CumulativeDistributionTransform` have no mapping doc, failed to reference from alias `torch.distributions.CumulativeDistributionTransform`
>   > Alias Warning: api `torch.distributions.transforms.SoftplusTransform` have no mapping doc, failed to reference from alias `torch.distributions.SoftplusTransform`
>   > Alias Warning: api `torch.igamma` have no mapping doc, failed to reference from alias `torch.special.gammainc`
>   > Alias Warning: api `torch.igammac` have no mapping doc, failed to reference from alias `torch.special.gammaincc`
>   > Alias Warning: api `torch.nn.modules.Linear` have no mapping doc, failed to reference from alias `torch.nn.linear`
>   > Alias Warning: api `torch.nn.modules.RNNCellBase` have no mapping doc, failed to reference from alias `torch.nn.RNNCellBase`
>   > Alias Warning: api `torch.nn.modules.linear.Linear` have no mapping doc, failed to reference from alias `torch.nn.linear`
>   > Alias Warning: api `torch.nn.modules.rnn.RNNCellBase` have no mapping doc, failed to reference from alias `torch.nn.RNNCellBase`
>   > Alias Warning: api `torch.nn.parameter.UninitializedParameter` have no mapping doc, failed to reference from alias `torch.nn.UninitializedParameter`
>   > Alias Warning: api `torch.selu_` have no mapping doc, failed to reference from alias `torch.nn.functional.selu_`
>   > Alias Warning: api `torch.threshold_` have no mapping doc, failed to reference from alias `torch.nn.functional.threshold_`
>   > Alias Warning: api `torch.torch.int32` have no mapping doc, failed to reference from alias `torch.int32`
>   > Alias Warning: api `transformers.generation.logits_process.LogitsProcessorList` have no mapping doc, failed to reference from alias `transformers.generation.LogitsProcessorList`
>   > Alias Warning: api `transformers.modeling_utils.PreTrainedModel` have no mapping doc, failed to reference from alias `transformers.PreTrainedModel`
>   > Alias Warning: api `transformers.utils.logging.get_logger` have no mapping doc, failed to reference from alias `transformers.logging.get_logger`

这个`torch.nn.linear`，在PaConvert中写错了，所以没匹配上。其他的都需要补充映射文档，除了 `transformers.*` 系列的API，其他的根据Matcher你这边补充一下文档吧：**缺失API映射文档补充**这一块工作，可能还缺少较多其他的要一起做（凡是有Matcher都需要有文档）

**Comment by RedContritio** [comment]
> 这个`torch.nn.linear`，在PaConvert中写错了，所以没匹配上。其他的都需要补充映射文档，除了 `transformers.*` 系列的API，其他的根据Matcher你这边补充一下文档吧：**缺失API映射文档补充**这一块工作，可能还缺少较多其他的要一起做（凡是有Matcher都需要有文档）

这部分改动比较大，准备等先合入了再追加补充，避免和其他 PR 冲突。
`torch.nn.Linear` 已更新。

**Review by zhwesky2010 — APPROVED** [review]

**Pull request merged**